File: freight-triage-polyglot-ledger-recovery/environment/reconcile/internal/normx/registry.go
package normx

// Registry lists every normalizer in catalogue order.
func Registry() []Normalizer {
	return []Normalizer{
		{Name: "upper_ascii", Apply: UpperAscii},
		{Name: "lower_ascii", Apply: LowerAscii},
		{Name: "trim_edges", Apply: TrimEdges},
		{Name: "collapse_spaces", Apply: CollapseSpaces},
		{Name: "strip_non_alnum", Apply: StripNonAlnum},
		{Name: "dash_to_underscore", Apply: DashToUnderscore},
		{Name: "pad_left_eight", Apply: PadLeftEight},
		{Name: "reverse_bytes", Apply: ReverseBytes},
		{Name: "rot13_letters", Apply: Rot13Letters},
		{Name: "digits_only", Apply: DigitsOnly},
	}
}

File: freight-triage-polyglot-ledger-recovery/environment/native/src/norm/registry.cpp
#include "freight/normalize.h"

namespace freight {

const std::vector<Normalizer>& normalizerRegistry() {
  static const std::vector<Normalizer> registry = {
      Normalizer{"upper_ascii", &norm_upper_ascii},
      Normalizer{"lower_ascii", &norm_lower_ascii},
      Normalizer{"trim_edges", &norm_trim_edges},
      Normalizer{"collapse_spaces", &norm_collapse_spaces},
      Normalizer{"strip_non_alnum", &norm_strip_non_alnum},
      Normalizer{"dash_to_underscore", &norm_dash_to_underscore},
      Normalizer{"pad_left_eight", &norm_pad_left_eight},
      Normalizer{"reverse_bytes", &norm_reverse_bytes},
      Normalizer{"rot13_letters", &norm_rot13_letters},
      Normalizer{"digits_only", &norm_digits_only},
  };
  return registry;
}

}  // namespace freight
